sympy/sets/sets.py

```diff
@@ -2457,7 +2457,10 @@ def simplify_intersection(args):
                 # with t. Returns the newly intersected set otherwise
 
                 if new_set is not None:
-                    new_args = (args - {s, t}).union({new_set})
+                    if isinstance(new_set, tuple):
+                        new_args = (args - {s, t}).union(new_set)
```

Comment:
which `intersection_sets` is returning a tuple? That should maybe be fixed?

Reply:
The new one... If you have a better idea, I'd be happy to implement it. The thing is that with the new code one would like to return an `Intersection`/two `Interval`s for the case when one have rewritten one of the bounds (but did not manage to merge it into one).

Reply:
Ahh, it's `new_set` not `new_args` that is a tuple. I would be more worried about the former.